api/v1beta1/openstackbaremetalset.go

```diff
@@ -76,11 +74,8 @@ func VerifyBaremetalStatusBmhRefs(
 		}
 
 		if !found {
-			err := fmt.Errorf("existing BaremetalHost \"%s\" not found for OpenStackBaremetalSet %s.  "+
-				"Please check BaremetalHost resources and re-add \"%s\" to continue",
-				bmhStatus.BmhRef, instance.Name, bmhStatus.BmhRef)
-
-			return err
```

Comment:
We explicitly added this logic to OSPdO because of a particularly nasty state that the various CRs involved could encounter -- one that caused compute-to-host assignments to thrash endlessly (along with continually reprovisioning).  I think that was a vulnerability of the way we chose hosts and only had a `count` field, which I believe is different in RHOSO now that we use maps.  So if deleting a BMH seems to be something we can handle, this change is fine with me.

Reply:
It seems to work fine with my testing i.e. deleting provisioned bmhs and it provisioning from available bmhs. This would make the compute replacement process we have in OSPRH 17.1 pretty simple in OSPRH 18.